File: appMain/src/main/java/com/cartlc/tracker/act/ListEntriesActivity.java
package com.cartlc.tracker.act;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;

import com.cartlc.tracker.R;
import com.cartlc.tracker.etc.PrefHelper;

import butterknife.BindView;
import butterknife.ButterKnife;

public class ListEntriesActivity extends AppCompatActivity {

    @BindView(R.id.list_entries) RecyclerView mEntriesList;
    @BindView(R.id.toolbar)      Toolbar      mToolbar;

    EntryListAdapter mEntryListAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_list_entries);
        ButterKnife.bind(this);
        mEntryListAdapter = new EntryListAdapter(this);
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
        linearLayoutManager.setAutoMeasureEnabled(true);
        mEntriesList.setLayoutManager(linearLayoutManager);
        mEntriesList.setAdapter(mEntryListAdapter);
        setSupportActionBar(mToolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }

    @Override
    protected void onResume() {
        super.onResume();
        mEntryListAdapter.onDataChanged();
        setTitle(PrefHelper.getInstance().getProjectName());
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.list_entries, menu);
        return true;
    }
    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int itemId = item.getItemId();
        if (itemId == android.R.id.home) {
            finish();
        } else if (itemId == R.id.edit) {
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public boolean onPrepareOptionsMenu(Menu menu) {
        MenuItem item = menu.getItem(R.id.edit);
        item.setVisible(true);
        return super.onPrepareOptionsMenu(menu);
    }
}

File: appMain/src/main/java/com/cartlc/tracker/fresh/ui/listentries/ListEntriesActivity.kt
/*
 * Copyright 2019, FleetTLC. All rights reserved
 */

package com.cartlc.tracker.fresh.ui.listentries

import android.os.Bundle
import android.view.MenuItem
import com.cartlc.tracker.R
import com.cartlc.tracker.fresh.ui.app.factory.FactoryController
import com.cartlc.tracker.fresh.ui.app.factory.FactoryViewMvc
import com.cartlc.tracker.fresh.ui.base.BaseActivity
import kotlinx.android.synthetic.main.activity_list_entries.*

class ListEntriesActivity: BaseActivity(), ListEntriesController.Listener {

    private val factoryViewMvc: FactoryViewMvc
        get() = componentRoot.factoryViewMvc
    private val factoryController: FactoryController
        get() = componentRoot.factoryController

    private lateinit var controller: ListEntriesController

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_list_entries)
        setSupportActionBar(toolbar_list)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)

        val viewMvc = factoryViewMvc.allocListEntriesViewMvc(content)
        controller = factoryController.allocListEntriesController(boundAct, viewMvc)
        controller.listener = this
        content.addView(viewMvc.rootView)
    }

    override var titleString: String?
        get() = title.toString()
        set(value) { title = value }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val itemId = item.itemId
        if (itemId == android.R.id.home) {
            finish()
        }
        return super.onOptionsItemSelected(item)
    }

}